packages/rum-core/src/domain/rumEventsCollection/action/trackClickActions.ts

```diff
@@ -113,39 +113,38 @@ export function trackClickActions(
 
     const startClocks = clocksNow()
 
-    const click = newClick(lifeCycle, history, trackFrustrations, {
+    const click = newClick(lifeCycle, history, selectionChanged, {
       name,
       event,
       startClocks,
     })
 
-    if (trackFrustrations && (!currentRageClickChain || !currentRageClickChain.tryAppend(click))) {
-      currentRageClickChain = createRageClickChain(click)
+    if (trackFrustrations && (!currentClickChain || !currentClickChain.tryAppend(click))) {
+      currentClickChain = createClickChain(click)
     }
 
     const { stop: stopWaitPageActivityEnd } = waitPageActivityEnd(
       lifeCycle,
       domMutationObservable,
       configuration,
       (pageActivityEndEvent) => {
-        if (!pageActivityEndEvent.hadActivity) {
-          // If it has no activity, consider it as a dead click.
-          // TODO: this will yield a lot of false positive. We'll need to refine it in the future.
-          if (trackFrustrations) {
-            click.addFrustration(FrustrationType.DEAD_CLICK)
-            click.stop()
-          } else {
-            click.discard()
-          }
-        } else if (pageActivityEndEvent.end < startClocks.timeStamp) {
+        if (pageActivityEndEvent.hadActivity && pageActivityEndEvent.end < startClocks.timeStamp) {
           // If the clock is looking weird, just discard the click
           click.discard()
-        } else if (trackFrustrations) {
-          // If we track frustrations, let's stop the click, but validate it later
-          click.stop(pageActivityEndEvent.end)
         } else {
-          // Else just validate it now
-          click.validate(pageActivityEndEvent.end)
+          click.stop(pageActivityEndEvent.hadActivity ? pageActivityEndEvent.end : undefined)
+
+          // Validate or discard the click only if we don't track frustrations. It'll be done when
+          // the click chain is finalized.
+          if (!trackFrustrations) {
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review (if it exists :^)  -->
**👏 praise:** ‏Once this check will be removed this part of the code will be super neat :D Nice!